File: android/app/src/main/java/com/example/demo/DemoScreen.kt
package com.example.demo

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.GridItemSpan
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.example.demo.domain.DemoViewModel
import com.example.demo.ui.components.RegionOverflowMenu
import com.example.demo.ui.components.SubscriptionCard

@Composable
fun DemoScreen(
    modifier: Modifier = Modifier,
    viewModel: DemoViewModel = hiltViewModel(),
) {
    val uiState by viewModel.uiState.collectAsStateWithLifecycle()

    if (uiState.isLoading) {
        Box(
            modifier = modifier.fillMaxSize(),
            contentAlignment = Alignment.Center,
        ) {
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(16.dp),
            ) {
                CircularProgressIndicator()
                Text(
                    text = "Initializing feature flags...",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }
        }
    } else if (uiState.initializationError) {
        Box(
            modifier = modifier.fillMaxSize(),
            contentAlignment = Alignment.Center,
        ) {
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(16.dp),
            ) {
                Text(
                    text = "Failed to initialize feature flags",
                    style = MaterialTheme.typography.headlineSmall,
                    color = MaterialTheme.colorScheme.error,
                )
                Text(
                    text = "Please check your configuration and try again",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }
        }
    } else {
        LazyVerticalGrid(
            columns = GridCells.Adaptive(minSize = 280.dp),
            horizontalArrangement = Arrangement.spacedBy(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp),
            contentPadding = PaddingValues(16.dp),
            modifier = modifier.fillMaxSize(),
        ) {
            item(span = { GridItemSpan(maxLineSpan) }) {
                Box(
                    modifier = Modifier.fillMaxWidth(),
                ) {
                    Text(
                        text = uiState.title,
                        style = MaterialTheme.typography.headlineLarge,
                        fontWeight = FontWeight.Bold,
                        textAlign = TextAlign.Center,
                        color = MaterialTheme.colorScheme.onSurface,
                        modifier =
                            Modifier
                                .fillMaxWidth()
                                .padding(bottom = 24.dp),
                    )

                    // Overflow menu
                    RegionOverflowMenu(
                        currentRegion = uiState.currentRegion,
                        showRegionMenu = uiState.showRegionMenu,
                        onToggleMenu = { viewModel.toggleRegionMenu() },
                        onSelectRegion = { regionCode -> viewModel.selectRegion(regionCode) },
                        onClearOverride = { viewModel.clearRegionOverride() },
                        modifier = Modifier.align(Alignment.TopEnd),
                    )
                }
            }

            items(uiState.subscriptionPlans) { plan ->
                SubscriptionCard(
                    plan = plan,
                    onSelectPlan = {
                        // Handle plan selection
                    },
                )
            }
        }
    }
}

File: ios/ConfidenceDemoApp/UI/DemoScreen.swift
import SwiftUI

struct DemoScreen: View {
    @EnvironmentObject var viewModel: DemoViewModel
    @Environment(\.materialTheme) var theme

    private let gridColumns = [
        GridItem(.adaptive(minimum: 280), spacing: 16)
    ]

    var body: some View {
        Group {
            if viewModel.uiState.isLoading {
                LoadingView()
            } else if viewModel.uiState.initializationError {
                ErrorView()
            } else {
                MainContentView()
            }
        }
        .materialDesignTheme()
    }

    private func LoadingView() -> some View {
        VStack(spacing: 16) {
            ProgressView()
                .scaleEffect(1.2)

            Text("Initializing feature flags...")
                .font(.body)
                .foregroundColor(theme.onSurfaceVariant)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(theme.surface)
    }

    private func ErrorView() -> some View {
        VStack(spacing: 16) {
            Text("Failed to initialize feature flags")
                .font(.title2)
                .fontWeight(.bold)
                .foregroundColor(theme.error)

            Text("Please check your configuration and try again")
                .font(.body)
                .foregroundColor(theme.onSurfaceVariant)
                .multilineTextAlignment(.center)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(theme.surface)
    }

    private func MainContentView() -> some View {
        ScrollView {
            LazyVGrid(columns: gridColumns, spacing: 16) {
                // Header with title and region menu
                HeaderView()
                    .gridCellColumns(2)

                // Subscription cards
                ForEach(viewModel.uiState.subscriptionPlans) { plan in
                    SubscriptionCard(plan: plan) {
                        // Handle plan selection
                        print("Selected plan: \(plan.name)")
                    }
                }
            }
            .padding(16)
        }
        .background(theme.surface)
        .onTapGesture {
            // Close region menu when tapping outside
            if viewModel.uiState.showRegionMenu {
                viewModel.toggleRegionMenu()
            }
        }
    }

    private func HeaderView() -> some View {
        VStack(spacing: 24) {
            HStack {
                Spacer()

                Text(viewModel.uiState.title)
                    .font(.largeTitle)
                    .fontWeight(.bold)
                    .foregroundColor(theme.onSurface)
                    .multilineTextAlignment(.center)

                Spacer()

                // Region overflow menu
                RegionOverflowMenu(
                    currentRegion: viewModel.uiState.currentRegion,
                    showRegionMenu: viewModel.uiState.showRegionMenu,
                    onToggleMenu: {
                        viewModel.toggleRegionMenu()
                    },
                    onSelectRegion: { regionCode in
                        viewModel.selectRegion(regionCode)
                    },
                    onClearOverride: {
                        viewModel.clearRegionOverride()
                    }
                )
            }
            .frame(maxWidth: .infinity)
        }
    }
}

extension GridItem {
    static func adaptive(minimum: CGFloat, spacing: CGFloat = 0) -> GridItem {
        GridItem(.adaptive(minimum: minimum), spacing: spacing)
    }
}

extension View {
    func gridCellColumns(_ count: Int) -> some View {
        // SwiftUI doesn't have direct equivalent to Android's GridItemSpan
        // This is a placeholder for grid spanning functionality
        self
    }
}